lib/oauth.js

```diff
@@ -532,3 +540,44 @@ internals.location = function (request, protocol, location) {
     var host = request.info.host || (info.host + ':' + info.port);
     return protocol + '://' + host + request.path;
 };
+
+
+// Provide own QS implementation for cross node version support
+
+internals.encodePrimitive = function (value) {
+
+    var type = typeof value;
+    if (type === 'boolean') {
```

Comment:
I would stay away from micro optimizations. This code should go away once this no longer supports node 0.10 because we can use the querystring encode override option in 0.12 and io.